redis/src/main/scala/zio/redis/Interpreter.scala

```diff
@@ -225,6 +239,99 @@ trait Interpreter {
               sets.get(key).map(_.fold(Replies.EmptyArray)(Replies.array(_))),
               STM.succeedNow(Replies.WrongType)
             )
+          case api.Sets.SRandMember.name =>
+            val key = input.head.asString
+            STM.ifM(isSet(key))(
+              {
+                val maybeCount = input.tail.headOption.map(b => b.asString.toLong)
+                for {
+                  set     <- sets.getOrElse(key, Set.empty[String])
+                  asVector = set.toVector
+                  res     <- maybeCount match {
+                           case None             =>
+                             selectOne[String](asVector, randomPick).map(maybeValue =>
+                               maybeValue.map(RespValue.bulkString).getOrElse(RespValue.NullValue)
+                             )
+                           case Some(n) if n > 0 => selectN(asVector, n, randomPick).map(Replies.array)
+                           case Some(n) if n < 0 =>
+                             selectNWithReplacement(asVector, -1 * n, randomPick).map(Replies.array)
+                           case Some(0)          => STM.succeedNow(RespValue.NullValue)
+                         }
+                } yield res
+              },
+              STM.succeedNow(Replies.WrongType)
+            )
+          case api.Sets.SRem.name        =>
+            val key = input.head.asString
```

Comment:
I presume we're safe here as the client doesn't allow chunks to be empty, but please double check that.